tests/components/greeneye_monitor/conftest.py

```diff
@@ -45,7 +40,7 @@ def assert_temperature_sensor_registered(
 ):
     """Assert that a temperature sensor entity was registered properly."""
     sensor = assert_sensor_registered(hass, serial_number, "temp", number, name)
-    assert sensor.original_device_class == DEVICE_CLASS_TEMPERATURE
+    assert sensor.original_device_class is SensorDeviceClass.TEMPERATURE
```

Comment:
I think we should consider adjusting the type hint here. Maybe we need to create a generic type which encompases all the possible ***DeviceClass enums?
https://github.com/home-assistant/core/blob/f55668ff40c4c0a53411e53d6334e51d64990867/homeassistant/helpers/entity_registry.py#L113

cc @scop

Reply:
@epenet Yes I think that would be a good idea.  A generic class could be a superclass of the rest and then put in the type hints as `DeviceClass | str | None`, and maybe eventually `DeviceClass | None`.

But what does that change here in this PR?